scheduler/scheduler.go

```diff
@@ -63,35 +63,52 @@ func (p *PortResourcePool) ReserveIPPort(ip string, port int64, protocol string)
 		// if ip can't be found and it is not 0.0.0.0,  reserve on the ghost map
 		if ip != defaultIP {
 			if _, ok := ghostMap[ip]; !ok {
-				ghostMap[ip] = map[int64]bool{}
+				ghostMap[ip] = map[int64]string{}
 				logrus.Infof("Creating ghost map for IP %v on protocol %v", ip, protocol)
-				ghostMap[ip][port] = true
+				ghostMap[ip][port] = instanceUUID
 				logrus.Infof("Port %v is reserved for IP %v for ghost map on protocol %v", port, ip, protocol)
 				return nil
 			}
-			if !ghostMap[ip][port] {
-				ghostMap[ip][port] = true
+			if ghostMap[ip][port] == "" {
+				ghostMap[ip][port] = instanceUUID
 				logrus.Infof("Port %v is reserved for IP %v for ghost map on protocol %v", port, ip, protocol)
 				return nil
 			}
+			// check if the phase is instance.start and the instance is allocated
+			if phase == "instance.start" {
```

Comment:
You should not need this phase check. Simply, if the instance reserving the port already has it reserved, return nil.